src/deluge/model/instrument/kit.cpp

```diff
@@ -1172,6 +1177,52 @@ void Kit::receivedNoteForDrum(ModelStackWithTimelineCounter* modelStack, MIDIDev
 	}
 }
 
+void Kit::possiblySetSelectedDrumAndRefreshUI(Drum* thisDrum) {
+	if (midiEngine.midiSelectKitRow) {
+		//is the drum for the note received the same as the currently selected drum in the kit?
+		//if no, update selected drum
+		if (selectedDrum != thisDrum) {
+			selectedDrum = thisDrum;
+
+			//is the instrument clip for this kit the currently selected clip? (could be selected by holding clip in song)
+			//only need to potentially refresh the kit clip if we're actively in that clip
+			//we use this function from midi follow because it was designed to detect clip selections in song / arranger view
+			//and when a clip is selected in song / arranger view, you will need to refresh the mod led indicators
+			//and if using midi follow you may need to send updated midi feedback (because parameter context has changed)
+			Clip* clip = midiFollow.getClipForMidiFollow();
```

Comment:
refactor - if this is being used outside midi follow, and doesn't need the midi follow object, then it shouldn't belong to the midi follow class. If it does rely on the midi follow object then it shouldn't be used here where there's no control on the midifollow internal variables.

Using it like this is likely to cause future issues since any changes to how midi follow clips are selected will also impact kit clip selection, and that won't be obvious to look at when reviewing changes to midi follow